### PR TITLE
Do not require Confluent libraries for Kafka connector if Protobuf features are not being used

### PR DESCRIPTION
**Pull request by hashhar**
## Description

Fixes https://github.com/trinodb/trino/issues/17299

## Additional context and related issues

Add product tests for Kafka connector with Confluent licensed libraries

The older approach was to copy the Confluent licensed libraries to the
Kafka connector classpath for all product environments extending from
`Kafka`. This meant that it was not possible to detect changes which
introduce a hard-dependency on the Confluent licensed libraries for
Kafka connector startup.

02cc332319a00e8fec5096921f96cb9e0aceacc1 inadvertently made it so that
the Confluent libraries became required for the Kafka connector to start
up and was not caught by tests because all the product test environments
already had the Confluent libraries present on the Kafka connector
classpath.

This commit instead introduces a new product test environment
`EnvMultinodeKafkaConfluentLicense` which is the only product test
environment that copies the Confluent licensed libraries onto the Kafka
connector classpath. There is also a new test group
`kafka_confluent_license` which now includes all the tests for the
functionality which requires the Confluent licensed libraries.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Kafka
* Fix failure of server startup when a Kafka catalog is present. ({issue}`17299`)
```

### CODE REVIEW

**Comment by hashhar** [comment]
the first 2 commits are alternative solutions for the same problem - please suggest which is better.

**Review by kokosing — APPROVED** [review]
Nice!

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeKafkaConfluentLicense.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+/**
+ * {@link EnvMultinodeKafkaConfluentLicense} is intended to be the only Kafka product test environment which copies the non-free Confluent License libraries to the Kafka connector
+ * classpath to test functionality which requires those classes.
+ * The other {@link Kafka} environments MUST NOT copy these jars otherwise it's not possible to verify the out of box Trino setup which doesn't ship with the Confluent licensed
+ * libraries.
+ */
+@TestsEnvironment
+public final class EnvMultinodeKafkaConfluentLicense
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeKafkaConfluentLicense.java:44]: nit : The name is a bit misleading, initial assumption is something like we add a some sort of license file or license string for this environment. How about `EnvMultinodeConfluentKafka` - though `Kafka` image still refers to a confluent image - we could mention the new environment comes with non-free Confluent License libraries

**Review by Praveen2112 — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaAvroReadsSmokeTest.java [hunk 2]
```diff
@@ -154,13 +154,8 @@ public static Object[][] catalogs()
         };
     }
 
-    private static final class KafkaCatalog
+    private record KafkaCatalog(String catalogName, String topicNameSuffix, boolean columnMappingSupported, MessageSerializer messageSerializer)
```
Comment (wendigo) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaAvroReadsSmokeTest.java:157]: can we make it multiline?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaProtobufReadsSmokeTest.java [hunk 3]
```diff
@@ -137,13 +137,8 @@ public static Object[][] catalogs()
         };
     }
 
-    private static final class KafkaCatalog
+    private record KafkaCatalog(String catalogName, String topicNameSuffix, boolean columnMappingSupported, MessageSerializer messageSerializer)
```
Comment (wendigo) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaProtobufReadsSmokeTest.java:156]: multiline?

**Review by adamjshook — APPROVED** [review]
LGTM, well done! Thanks

**Pull request merged**